File: Programmers/Python/실패율.py
def solution(N, stages):
    res = []
    v = []
    sum = [0 for _ in range(1,N+1)]
    sum.append(0)
    mp = {} # int, int
    total = len(stages)
    
    for i in range(1,N+2):
        mp[i]=0
        
    for x in stages:
        mp[x]+=1
    
    for i in range(1,N+1):
        sum[i] = sum[i-1] + mp[i]
        
    for i in range(1,N+1):
        dv = total - sum[i-1]
        f = 0
        if dv!=0:
            f = mp[i] / dv
        v.append((f,i))
            
    v.sort(key = lambda x : (-x[0], x[1]))
    
    for x in v:
        res.append(x[1])
    return res

File: Programmers/C++/실패율.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
#define FOR(i,n) for(int i=0;i<(n);++i)
#define FORS(i,s,n) for(int i=(s);i<(n);++i)
#define MAX 505
using namespace std;

int mp[MAX],sum[MAX];

struct node {
    int idx;
    double score;
    
    bool operator < (const node& arg) const {
        if(score!=arg.score) return score>arg.score;
        return idx<arg.idx;
    }
};
vector<node> v;

vector<int> solution(int N, vector<int> stages) {
    vector<int> res;
    
    int total = stages.size();
    
    for(auto x: stages) mp[x]++;
    FORS(i,1,N+1) sum[i] = sum[i-1] + mp[i];
    
    FORS(i,1,N+1) {
        int dv = total-sum[i-1];
        double f;
        dv==0? f=0 : f=(double)mp[i]/dv;
        v.push_back({i,f});
    }
    sort(v.begin(),v.end());
    for(auto x: v) res.push_back(x.idx);
    return res;
}
